### PR TITLE
Feature/recent opened

### PR DESCRIPTION
**Pull request by emanlodovice**
Change log:
* add recently opened feature using localstorage
* fix version indicated in package.json

### CODE REVIEW

File: src/components/RecentlyOpened.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          path,
+          display: parts[parts.length - 1]
+        };
+      });
+    });
+  },
+
+  methods: {
+    ...mapActions(['getRecentlyOpened'])
+  }
```
Comment (arnellebalane) [inline src/components/RecentlyOpened.vue:36]: @emanlodovice Can we make recently opened array included in the store's state? Then the `created` method can then become a computed property based on the state.

Comment (emanlodovice) [reply src/components/RecentlyOpened.vue]: thanks for the review. updated @arnellebalane . please check.

File: src/store.js [hunk 2]
```diff
@@ -36,6 +37,10 @@ const mutations = {
 
   deleteKey(state, key) {
     state.data = state.data.filter(item => item.key !== key);
+  },
+
+  setRecentlyOpenedPaths(state, paths) {
+    state.recentlyOpenedPaths = [...paths];
```
Comment (arnellebalane) [inline src/store.js:43]:
```suggestion
    state.recentlyOpenedPaths = paths;
```

No need to spread and create a new array.

File: src/store.js [hunk 3]
```diff
@@ -126,6 +131,20 @@ const actions = {
...
+  populateRecentlyOpened({ commit }) {
+    const storedData = window.localStorage.getItem('recentlyOpened');
+    if (storedData) {
+      commit('setRecentlyOpenedPaths', JSON.parse(storedData));
+    }
+  },
+
+  addToRecentlyOpened({ commit, state }, path) {
+    const newPaths = state.recentlyOpenedPaths.filter(p => p != path);
+    newPaths.unshift(path);
```
Comment (arnellebalane) [inline src/store.js:145]:
```suggestion
    const newPaths = [path, ...state.recentlyOpenedPaths.filter(p => p != path)];
```

How about something like this? 😅

File: src/components/RecentlyOpened.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  created() {
+    this.populateRecentlyOpened();
+  },
+
+  computed: {
+    ...mapState(['recentlyOpenedPaths']),
+
+    paths() {
+      const pathsToDisplay = [...this.recentlyOpenedPaths].splice(0, 2); // show only 2 paths
```
Comment (arnellebalane) [inline src/components/RecentlyOpened.vue:28]: @emanlodovice Why not `slice` the `recentlyOpenedPaths` array directly instead?

Comment (emanlodovice) [reply src/components/RecentlyOpened.vue:28]: because it changes the value stored on state.

File: src/store.js (hunk 3 above)
Comment (emanlodovice) [reply src/store.js]: updated

File: src/components/RecentlyOpened.vue (hunk 4 above)
Comment (emanlodovice) [reply src/components/RecentlyOpened.vue]: updated

File: src/store.js (hunk 2 above)
Comment (emanlodovice) [reply src/store.js]: updated

File: src/components/RecentlyOpened.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="recently-opened">
+    <span>Recently Opened:</span>
+    <div class="recent-items">
+      <a v-for="path in paths" v-bind:title="path.path" :key="path.path" @click="$emit('selected', path.path)">
```
Comment (arnellebalane) [inline src/components/RecentlyOpened.vue:5]:
```suggestion
      <a v-for="path in paths" :title="path.path" :key="path.path" @click="$emit('selected', path.path)">
```

**Pull request merged**